Uniques040

MysticJay opened the pull request:
This PR extends UNIQUES to allow keeping track of portals that have been scanned for "Scout Controller" (scoutControlled) or visited by a "Drone" (dronedVisited).

In general the user has to mark scoutControlled or droneVisited portals manually. Additionally UNIQUES will analyze the COMM on ALL and ALERTS (requires a core change to CHAT.JS) for messages that indicate a previous actions withthe portal. The information will be added immediately to the portal-list to those portals visible on the map. For unresolved portals the information (LatLonE6) will be stored in a separate list called missedPortals. As soon as one of the missed portals appears on the map the information will be added to portal-list as well.

To improve the hitrate, when resolving the information the plugin Cache-Portals-On-Map and Offle can assist to resolve LatLonE6 to guid.

UNIQUES now has a dialog that allows to jump to those locations listed in missedLatLngs. Also it allows to export/backup and iport/restore UNIQUES data to/from a local file.

c-schmitz committed:
First commit

c-schmitz committed:
Fixed linebreaks

c-schmitz committed:
Create README.md

c-schmitz committed:
-Added import - will never delete, only update portals

c-schmitz committed:
Updated README

koljanowak committed:
Fixed include/match header lines for the new Intel URL. (#1)

MysticJay committed:
Adding support for scouted and droned portals

Routines and data cretaed.

MysticJay committed:
Highlighters

Added Highlighters
description
added new text to comm analysis
minor cosmetics

MysticJay committed:
sorting portalslist fixed

sorting scouted or droned caused an empty list, as old uniques might not have the related fields.

MysticJay committed:
sorting uniques fixed

Some combinations should not happen, but they do.
fixed too.

MysticJay committed:
Uniques-Export

Merge Project
https://github.com/c-schmitz/iitc-unique-export
Move code to uniques.js.

MysticJay committed:
0.3.9 beta 1

- integrated uniques im-/export (@c-schmitz)
dialog to export or merge uniques lists.
- integrated missedLatLngs (@sp00x)
uniques will remember portals that were mentioned in chat and update them when visible in map automatically.

todo:
- widen portallist dialog
- add dialog to list pending portal updates
- option to choose different styles for highlights.

MysticJay committed:
UNIQUES 0.3.9beta2

- Added list to manage missed portals.
Advise: Please delete from localStorage "plugin-uniques-missedLatLngs"
the old structure does not support the dialog (but it will not break the missedLatLngs in general).

WiP:
- option to choose different styles for highlights.

ToDo:
- widen portallist dialog

MysticJay committed:
UNIQUES 0.3.9rc1

Feature complete
- renamed export/import to backup/restore
- merged the dialogs,
- removed the copy paste araea
- added missingLatLng List to dialog

MysticJay committed:
Cleanup

removing obsolete comments and disabled code

MysticJay committed:
PortalList-Dialog fixed

Added CSS to remove dialog max width limitation

MysticJay committed:
Typo fix

minor change, but will cause the list of already parsed messages to be rebuild.
Old values stay in localstore but can be deleted without negative impact.

MysticJay committed:
abandoned dialog

Removing abandoned dialog (copy/paste im/-export)

MysticJay committed:
Update uniques.js

less verbosity
fixed function call to new syntax

MysticJay committed:
Import fixed

- Import function fixed
- improved dialog

MysticJay committed:
Undoing indentation

changing indentation from tab to 2 spaces caused too many changes- undone

MysticJay committed:
indentation of headers

headers aligned

le-jeu commented on plugins/uniques.js:
```diff
@@ -33,878 +33,878 @@ window.plugin.uniques.contentHTML = null;
+				markup = plext.markup;
+
+		if(plext.plextType == 'SYSTEM_BROADCAST'
+		&& markup.length==5
```
the indentation here doesnt match

le-jeu commented on plugins/uniques.js:
```diff
@@ -552,7 +522,27 @@ window.plugin.uniques.highlighterDroned = {
 window.plugin.uniques.setupCSS = function() {
   $("<style>")
   .prop("type", "text/css")
-  .html('@include_string:uniques.css@')
```
either delete `uniques.css` or use it, using it sounds better

MysticJay replied:
While I have no idea where the old code was introduced, the latest commit moves the code back to uniques.css

MysticJay committed:
Bump version to 0.4.0

0.3.9 -> 0.4.0

MysticJay committed:
include uniques.css

moving code back to using uniques.css

MysticJay committed:
New visits in COMM

Added
"x deployed a Beacon on z"
"x deployed Fireworks on z"

modos189 commented on plugins/uniques.js:
```diff
@@ -510,45 +731,144 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+window.plugin.uniques.optExport = function() {
+	var data = localStorage['plugin-uniques-data'];
+	window.saveFile(data, 'IITC-uniques.json', 'application/json');
+	confirm ("IITC-Uniques exported to IITC-uniques.json");
```
What if I press Cancel? And what changes if I press OK?
I propose to remove this line, similar to Draw tools - no such window appears there.

![изображение](https://user-images.githubusercontent.com/1785196/105163986-9feda000-5b25-11eb-8b74-5dd0f195b153.png)

```suggestion
```

MysticJay committed:
removing confirm

duplicate dialog during export removed

Co-authored-by: Alexander Danilov <git@modos189.ru>

MysticJay committed:
unified highlighter names

Names for the highlighters were in conflic with history highlighters.

MysticJay committed:
UI fixes

UI-fixed
Scout Controller
Drone visited

johnd0e commented on plugins/uniques.css:
```diff
@@ -17,3 +17,7 @@
+
+.ui-dialog-portalslist {
+  max-width: none !important;
+}
```
This should not belong here

johnd0e replied:
?

MysticJay replied:
We have to widen the portal-list dialog to avoid a horizontal scroll bar.
it was fixed to 600px for no obvious reason.
This is the only place we found to be appropriate

le-jeu replied:
https://github.com/IITC-CE/ingress-intel-total-conversion/blob/67fcdf9adb8c079de6332742f12df7c99f4b91bb/core/style.css#L1036-L1044

Don't use `!important`
prefer use more specific rules like 
```css
.ui-dialog.ui-dialog-portallist {
  max-width: max-width: calc(100vw - 2px);
}
```

or open a discussion about the 600px

MysticJay replied:
trying that

johnd0e commented on plugins/uniques.js:
```diff
@@ -1,9 +1,8 @@
-// @version        0.2.4
-// @description    Allow manual entry of portals visited/captured. Use the 'highlighter-uniques' plugin to show the uniques on the map, and 'sync' to share between multiple browsers or desktop/mobile. It will try and guess which portals you have captured from COMM/portal details, but this will not catch every case.
-
+// @version        0.4.0
```
Why skip 0.3?

MysticJay replied:
0.3 was some intermediate release and I want to ensure that people get the update from release. so 0.4.0 was never used in either a draft or beta or RC and it will update.

johnd0e replied:
> 0.3 was some intermediate release

As I can see we never had such release on official site.

>  I want to ensure that people get the update from release.

Do you mean auto-update routines?
They also rely on `@updateURL`, so it's not enough to change version.

johnd0e commented on plugins/uniques.js:
```diff
@@ -60,9 +63,12 @@ window.plugin.uniques.onPortalDetailsUpdated = function() {
 
 window.plugin.uniques.onPublicChatDataAvailable = function(data) {
 	var nick = window.PLAYER.nickname;
+	let match = false;
```
var

johnd0e commented on plugins/uniques.js:
```diff
@@ -167,9 +218,13 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
-			captured = (uniqueInfo && uniqueInfo.captured) || false;
+			captured = (uniqueInfo && uniqueInfo.captured) || false,
+			scouted = (uniqueInfo && uniqueInfo.scouted) || false,
+			droned = (uniqueInfo && uniqueInfo.droned) || false;
```
Should be same names as in history.
Here and in numerous places below.

johnd0e replied:
Still unresolved. Properties, ids, ...

MysticJay replied:
code and properties changed:
- scouted -> scoutControlled
- droned ->droneVisited

converter implemented. It will add a version info to the uniques MAP.

johnd0e commented on plugins/uniques.js:
```diff
@@ -179,42 +234,39 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
-	plugin.uniques.sync(guid);
 }
-
+// Visted
```
Typo. And why do we need this comment?

johnd0e commented on plugins/uniques.js:
```diff
@@ -238,7 +290,7 @@ window.plugin.uniques.updateVisited = function(visited, guid) {
 	plugin.uniques.sync(guid);
 }
-
+// Captured
```
Needless?

MysticJay replied:
personal flavor

johnd0e commented on plugins/uniques.js:
```diff
@@ -262,6 +314,50 @@ window.plugin.uniques.updateCaptured = function(captured, guid) {
 	plugin.uniques.updateCheckedAndHighlight(guid);
 	plugin.uniques.sync(guid);
 }
+// Scouted
```
Needless?

MysticJay replied:
personal flavor

johnd0e commented on plugins/uniques.js:
```diff
@@ -262,6 +314,50 @@ window.plugin.uniques.updateCaptured = function(captured, guid) {
+	plugin.uniques.updateCheckedAndHighlight(guid);
+	plugin.uniques.sync(guid);
+}
+// Droned
```
Needless?

MysticJay replied:
personal flavor

johnd0e commented on plugins/uniques.js:
```diff
@@ -262,6 +314,50 @@ window.plugin.uniques.updateCaptured = function(captured, guid) {
+
+	if(scouted == uniqueInfo.scouted) return;
+
+	if (scouted) {
```
Space after `if`, here and everywhere.

johnd0e commented on plugins/uniques.js:
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+				if (!confirm("Please only confirm this if you know what you are doing!!\nAre you sure you want to save your Unique visits/captures back to IITC?")) return;
+				var data = JSON.parse(e.reader.result);
+
+				if (Object.keys(data).length) window.plugin.uniques.uniques = data;
```
```suggestion
				if (Object.keys(data).length) {
				  window.plugin.uniques.uniques = data;
				}
```

johnd0e commented on plugins/uniques.js:
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+	L.FileListLoader.loadFiles({accept:'application/json'})
+		.on('load',function (e) {
+			try {
+				if (!confirm("Please only confirm this if you know what you are doing!!\nAre you sure you want to save your Unique visits/captures back to IITC?")) return;
```
```suggestion
				var msgExit = 'Please only confirm this if you know what you are doing!\n'
				                    + 'Are you sure you want to save your Unique visits/captures back to IITC?';
				if (!confirm(msgExit)) {
				  return;
				}
```

(too long line)

johnd0e commented on plugins/uniques.js:
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+window.plugin.uniques.onPortalAdded = function(data) {
+	let guid = data.portal.options.guid;
+	let p = data.portal.options.data;
+	let id = p.latE6 + "," + p.lngE6;
```
var

johnd0e commented on plugins/uniques.js:
```diff
@@ -510,45 +731,143 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+window.plugin.uniques.removeOldParsedMsgs = function() {
+//	remove all timestamps older than 30 days
+	let old = (Date.now() - (30*24*60*60*1000)); //30 days in miliseconds
+	let count = 0;
```
var

johnd0e commented on plugins/uniques.js:
```diff
@@ -557,15 +876,30 @@ var setup = function() {
+
+	// add controls to toolbox
+
+	link = $("<a onclick=\"window.plugin.uniques.options();return false\" title=\"Manage UNIQUES\">Uniques</a>");
```
```suggestion
	var link = $('<a onclick="window.plugin.uniques.options();return false" title="Manage UNIQUES">Uniques</a>');
```

BUT!
Avoid js code in html attributes.
Replace it with jQuery listeners.
```suggestion
	var link = $('<a title="Manage UNIQUES">Uniques</a>')
	  .click(window.plugin.uniques.options);
```

johnd0e reviewed (changes requested):
Very brief review for now.

MysticJay committed:
Apply suggestions from code review done (@Johnd0e)

small fixes as proposed by @johnd0e

Co-authored-by: johnd0e <johnd0e@mail.ua>

MysticJay committed:
Code style fixes

fixing code style as requested by @johnd0e

johnd0e commented on plugins/uniques.js:
```diff
@@ -399,6 +496,48 @@ window.plugin.uniques.highlighter = {
+		var style = {};
+
+		if (uniqueInfo && uniqueInfo.scouted)
+		{ // scouted - no highlights
```
https://eslint.org/docs/rules/brace-style (1tbs)

MysticJay replied:
done

johnd0e commented on plugins/uniques.js:
```diff
@@ -399,6 +496,48 @@ window.plugin.uniques.highlighter = {
+		var style = {};
+
+		if (uniqueInfo && uniqueInfo.droned)
+		{ // droneded - no highlights
```
https://eslint.org/docs/rules/brace-style (1tbs)

MysticJay replied:
done

johnd0e commented on plugins/uniques.js:
```diff
@@ -411,50 +550,71 @@ window.plugin.uniques.setupContent = function() {
-	window.plugin.portalslist.fields.push({
-		title: "Visit",
+	window.plugin.portalslist.fields.push(
+		{title: "V/C",
```
https://eslint.org/docs/rules/brace-style (1tbs)

MysticJay replied:
reason: having the { in the same line as "title" allows to fold the section in the editor.
nevertheless moved the { back to previous line.

johnd0e commented on plugins/uniques.js:
```diff
@@ -487,16 +647,77 @@ window.plugin.uniques.setupPortalsList = function() {
-	});
-}
+//---------------------------------------------------------------------------
+		{title: "S",
```
`title` better move to next line, indented

MysticJay replied:
done

johnd0e commented on plugins/uniques.js:
```diff
@@ -487,16 +647,77 @@ window.plugin.uniques.setupPortalsList = function() {
+		},
+	},
+//---------------------------------------------------------------------------
+		{title: "D",
```
Same here

MysticJay replied:
done

MysticJay committed:
Code style

code style changes

MysticJay reviewed (commented):
reviewed the change requests.
some accepted, some delayed to next release.

MysticJay committed:
style fixes

MysticJay reviewed (commented):
code style fixes

johnd0e commented on plugins/uniques.js:
```diff
@@ -139,10 +167,10 @@ window.plugin.uniques.onPublicChatDataAvailable = function(data) {
 		&& markup[4][0] == 'TEXT'
 		&& markup[4][1].plain == ' has decayed') {
-		    // search for "Your Lx Resonator on y has decayed"
+				// search for "Your Lx Resonator on y has decayed"
```
Wrong indentation?

johnd0e commented on plugins/uniques.js:
```diff
@@ -154,8 +182,31 @@ window.plugin.uniques.onPublicChatDataAvailable = function(data) {
+			&& markup[0][1].plain == 'You were displaced as Scout Controller on '
+			&& markup[1][0] == 'PORTAL') {
+				// search for "You were displaced as Scout Controller on"
+				var portal = markup[1][1];
```
Indentation of `&&` does not much similar cases in upper code

MysticJay replied:
now matching the same broken indentation as the upper code

johnd0e commented on plugins/uniques.js:
```diff
@@ -154,8 +182,31 @@ window.plugin.uniques.onPublicChatDataAvailable = function(data) {
+				var portal = markup[1][1];
+				match = true;
+				plugin.uniques.setPortalAction(portal,'scouted');
+		}
+		}
```
Hm..
Hard to follow matching opening braces.

johnd0e commented on plugins/uniques.js:
```diff
@@ -179,84 +234,125 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
+		plugin.uniques.sync(guid);
+	} else { //guid not found, so add to missedLatLngs
+		if (!window.plugin.uniques.missedLatLngs[id])
+			window.plugin.uniques.missedLatLngs[id] = {portal:portal,action:{}};
```
Braces?

johnd0e commented on plugins/uniques.js:
```diff
@@ -179,84 +234,125 @@ window.plugin.uniques.updateCheckedAndHighlight = function(guid) {
+		}
+		uniqueInfo[action] = true;
+		// special handling for captured
+		if (action === 'captured') uniqueInfo.visited = true;
```
I'd prefer to see braces even in single-line conditions. Here and in numerous other places.

johnd0e commented on plugins/uniques.js:
```diff
@@ -411,50 +550,71 @@ window.plugin.uniques.setupContent = function() {
 		+ '<label><input type="checkbox" id="visited" onclick="window.plugin.uniques.updateVisited($(this).prop(\'checked\'))"> Visited</label>'
 		+ '<label><input type="checkbox" id="captured" onclick="window.plugin.uniques.updateCaptured($(this).prop(\'checked\'))"> Captured</label>'
+		+ '<label><input type="checkbox" id="scouted" onclick="window.plugin.uniques.updateScouted($(this).prop(\'checked\'))"> Scout controlled/label>'
+		+ '<label><input type="checkbox" id="droned" onclick="window.plugin.uniques.updateDroned($(this).prop(\'checked\'))"> Drone visited</label>'
```
Is it possible to replace inline js with jQuery listeners?

MysticJay replied:
@johnd0e: on TG "So if this is not clear to you - better do nothing"

johnd0e commented on plugins/uniques.js:
```diff
@@ -510,45 +733,149 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+	//console.log("portal added: %s %s %o", guid, id, data);
+	// check if portal is in missedLatLngs
+	if (window.plugin.uniques.missedLatLngs[id]) {
+		console.log("found portal guid for previously missed portal: %s -> %s (%o)", id, guid, data);
```
No excessive logging

MysticJay replied:
removed

johnd0e commented on plugins/uniques.js:
```diff
@@ -510,45 +733,149 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+	var list = '';
+	for (var item in mLL) {
+		var p = mLL[item].portal;
+		list = list + '<a onclick=\"map.setView(['+p.latE6/1E6+','+p.lngE6/1E6+'],15);\">'+p.name+'</a><br>';
```
```suggestion
		list = list + '<a onclick="map.setView(['+p.latE6/1E6+','+p.lngE6/1E6+'],15);">'+p.name+'</a><br>';
```
And is it possible to replace inline js with jQuery listeners?

MysticJay replied:
changed

johnd0e commented on plugins/uniques.js:
```diff
@@ -510,45 +733,149 @@ window.plugin.uniques.onMissionLoaded = function(data) {
+			+ window.plugin.uniques.genList()
+			+ '</div><hr>'
+			+ '<a onclick=\"window.plugin.uniques.optExport();return false\" title=\"Export portals\' unique info to IITC.\">Backup</a> / '
+			+ '<a onclick=\"window.plugin.uniques.optImport();return false\" title=\"Import portals\' unique info to IITC.\">Restore</a> Uniques'
```
```suggestion
			+ '<a onclick="window.plugin.uniques.optExport();return false" title="Export portals\' unique info to IITC.">Backup</a> / '
			+ '<a onclick="window.plugin.uniques.optImport();return false" title="Import portals\' unique info to IITC.">Restore</a> Uniques'
```
And is it possible to replace inline js with jQuery listeners?

MysticJay replied:
changed

johnd0e reviewed (changes requested):
Original code seems overcomplicated...

MysticJay committed:
propertyNames

ode and properties changed:

    scouted -> scoutControlled
    droned ->droneVisited

converter implemented. It will add a version info to the uniques MAP.

MysticJay committed:
Code-style fixes

MysticJay committed:
Update uniques.css

removing !important from CSS

le-jeu commented on plugins/uniques.js:
```diff
@@ -411,50 +588,71 @@ window.plugin.uniques.setupContent = function() {
 
 	window.plugin.portalslist.fields.push({
-		title: "Visit",
+		title: "V/C",
```
Visit/capture/scout will already be in portal-list (but drone/scanned aren't)
Don't you want to replace/edit the columns instead of duplicate them ?

johnd0e replied:
> Don't you want to replace/edit the columns instead of duplicate them ?

Reuse, not replace.
E.g. use cached uniques data when there are no info from intel.
And highlight text in cases when data differ (with more explanation in tooltip).

johnd0e commented on plugins/uniques.css:
```diff
@@ -18,6 +18,6 @@
-}
+.ui-dialog.ui-dialog-portallist {
+  max-width: max-width: calc(100vw - 2px);
+}
```
EOL expected

le-jeu commented on plugins/uniques.js:
```diff
@@ -411,50 +588,71 @@ window.plugin.uniques.setupContent = function() {
 	plugin.uniques.contentHTML = '<div id="uniques-container">'
 		+ '<label><input type="checkbox" id="visited" onclick="window.plugin.uniques.updateVisited($(this).prop(\'checked\'))"> Visited</label>'
 		+ '<label><input type="checkbox" id="captured" onclick="window.plugin.uniques.updateCaptured($(this).prop(\'checked\'))"> Captured</label>'
+		+ '<label><input type="checkbox" id="scoutControlled" onclick="window.plugin.uniques.updateScoutControlled($(this).prop(\'checked\'))"> Scout controlled/label>'
```
```suggestion
		+ '<label><input type="checkbox" id="scoutControlled" onclick="window.plugin.uniques.updateScoutControlled($(this).prop(\'checked\'))"> Scout controlled</label>'
```

le-jeu reviewed (changes requested):
`IngressUniqueExporter.user.js` needs to be removed
and `README.md` restored

you can simply rebase and drop the related commits to keep only those that changes `uniques.{css,js}`

Suburbanno commented:
@MysticJay Do you intend to continue this contribution? If not, could you close it?

MysticJay commented:
please stop this

MysticJay committed:
Update plugins/uniques.js

Co-authored-by: LeJeu <64744459+le-jeu@users.noreply.github.com>

MysticJay committed:
Update uniques.css
